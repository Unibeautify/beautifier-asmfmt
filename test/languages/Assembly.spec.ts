import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import { raw } from "../utils";

jest.setTimeout(10000);
describe("should successfully beautify Assembly files", () => {
  // tslint:disable:mocha-no-side-effect-code
  testFile("test.asm");
});

function testFile(fixtureFileName: string) {
  test(`should successfully beautify file ${fixtureFileName}`, () => {
    const text: string = fs
      .readFileSync(path.resolve(__dirname, `../fixtures/${fixtureFileName}`))
      .toString();
    const unibeautify = newUnibeautify();
    unibeautify.loadBeautifier(beautifier);
    return unibeautify
      .beautify({
        languageName: "Assembly",
        options: {},
        text,
      })
      .then(results => {
        expect(raw(results)).toMatchSnapshot();
      });
  });
}
