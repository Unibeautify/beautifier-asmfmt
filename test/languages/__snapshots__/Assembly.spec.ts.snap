// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully beautify Assembly files should successfully beautify file test.asm 1`] = `
Object {
  Symbol(raw): "	; https: // jameshfisher.com/2018/03/10/linux-assembly-hello-world.html

	global _start

	section .text

_start:
	mov     rax, 1; write(
	mov     rdi, 1; STDOUT_FILENO
	mov     rsi, msg; \\" Hello, world!\\\\n\\"
	mov     rdx, msglen; sizeof(\\"Hello, world!\\\\n\\")
	syscall ; )

	mov     rax, 60; exit(
	mov     rdi, 0; EXIT_SUCCESS
	syscall ; )

	section .rodata

msg:
	db \\"Hello, world!\\", 10

msglen:
	equ $ - msg
",
}
`;
